internal/goroutine/periodic.go

```diff
@@ -100,10 +103,14 @@ func NewPeriodicGoroutine(ctx context.Context, name, description string, interva
 // NewPeriodicGoroutineWithMetrics creates a new PeriodicGoroutine with the given handler. The context provided will propagate into
 // the executing goroutine and will terminate the goroutine if cancelled.
 func NewPeriodicGoroutineWithMetrics(ctx context.Context, name, description string, interval time.Duration, handler Handler, operation *observation.Operation) *PeriodicGoroutine {
-	return newPeriodicGoroutine(ctx, name, description, interval, handler, operation, glock.NewRealClock())
+	return newPeriodicGoroutine(ctx, name, description, interval, handler, operation, glock.NewRealClock(), nil)
+}
+
+func NewPeriodicGoroutineWithMetricsAndDynamicInterval(ctx context.Context, name, description string, handler Handler, getInterval getIntervalFunc, operation *observation.Operation) *PeriodicGoroutine {
+	return newPeriodicGoroutine(ctx, name, description, getInterval(), handler, operation, glock.NewRealClock(), getInterval)
 }
 
-func newPeriodicGoroutine(ctx context.Context, name, description string, interval time.Duration, handler Handler, operation *observation.Operation, clock glock.Clock) *PeriodicGoroutine {
+func newPeriodicGoroutine(ctx context.Context, name, description string, interval time.Duration, handler Handler, operation *observation.Operation, clock glock.Clock, getInterval getIntervalFunc) *PeriodicGoroutine {
```

Comment:
```suggestion
func newPeriodicGoroutine(ctx context.Context, name, description string, getInterval getIntervalFunc, handler Handler, operation *observation.Operation, clock glock.Clock) *PeriodicGoroutine {
```

Why not do this and pass in `func() time.Time { return interval}` in `NewPeriodicGoroutineWithMetrics`?

Reply:
updated